task/sensor/src/main.rs

```diff
@@ -128,16 +149,19 @@ fn main() -> ! {
     //
     sys_set_timer(Some(deadline), TIMER_MASK);
 
-    let data = mutable_statics::mutable_statics! {
-        static mut SENSOR_DATA: [Reading; NUM_SENSORS] = [|| Reading::Absent; _];
-    };
-
-    let nerrors = mutable_statics::mutable_statics! {
-        static mut SENSOR_NERRORS: [u32; NUM_SENSORS] = [|| 0; _];
+    let (has_data, data_value, data_time, nodata, nerrors) = mutable_statics::mutable_statics! {
+        static mut HAS_DATA: [bool; NUM_SENSORS] = [|| false; _];
+        static mut DATA_VALUE: [f32; NUM_SENSORS] = [|| f32::NAN; _];
```

Comment:
I admit I was briefly considering using `is_nan` to remove the need for your bool array.... ;-)

Reply:
Same, but I've now swapped stuff around to track the "most recent" reading (whether it's an error or data value), in 9cdbb2503ef75b495ffc0e576f04190106986ffa